packages/page-claims/src/useCounter.ts

```diff
@@ -0,0 +1,21 @@
+// Copyright 2017-2020 @polkadot/app-settings authors & contributors
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
+
+import { Option } from '@polkadot/types';
+import { EthereumAddress } from '@polkadot/types/interfaces';
+
+import { useAccounts, useApi, useCall } from '@polkadot/react-hooks';
+
+export default function useCounter (): number {
+  const { allAccounts } = useAccounts();
+  const { api, isApiReady } = useApi();
+
+  // Find accounts that need attest. They are accounts that
+  // - already preclaimed,
+  // - didn't sign the attest yet.
+  // `claims.preclaims` returns Some() for these accounts.
+  const needAttest = useCall<[Option<EthereumAddress>]>(isApiReady && api.query.claims?.preclaims?.multi, [allAccounts])?.filter((opt) => opt.isSome);
+
+  return needAttest ? needAttest.length : 0;
```

Comment:
```suggestion
  const counter = useCall<number>(isApiReady && api.query.claims?.preclaims?.multi, [allAccounts], {
    transform: (preclaims: Option<EthereumAddress>[]) =>
      preclaims.filter((opt) => opt.isSome).length
  });

  return counter || 0;
```

Technically since we do the exact same query in 2 places, we should have a hook that just extracts the accounts into an array (as above) and this one should just return the length of that hook.

Reply:
I extracted it in a `usePreclaimPolkadotAddresses` hook used locally. Is that what you were thinking about, or did you think about `@polkadot/react-hooks`?

Reply:
Locally is 100%. (Since it is only used here - sadly there are some local-only ones that slipped into react-hooks, just have not move the 1 or 2 yet)